pkg/chains/objects/objects.go

```diff
@@ -0,0 +1,196 @@
+package objects
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	"github.com/tektoncd/pipeline/pkg/client/clientset/versioned"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const (
+	pipelineRunKind = "PipelineRun"
+	taskRunKind     = "TaskRun"
+)
+
+type Annotation struct {
+	Err   error
+	Value string
+	Ok    bool
+}
+
+type Result struct {
+	Name  string
+	Value string
+}
+
+// Represents a generic K8s object
+// This isn't meant to be a final implementation, just one approach
+// Many of these methods can be abstracted away further
+type K8sObject interface {
+	GetName() string
+	GetNamespace() string
+	GetKind() string
+	GetAnnotation(annotation string) *Annotation
+	GetLatestAnnotation(annotation string) *Annotation
+	GetObject() interface{}
+	Patch(patchBytes []byte) error
+	GetResults() []Result          // TODO: Abstract this further to return any field in the status
+	GetServiceAccountName() string // TODO: Abstract this further to return any field in the spec
+}
+
+type TaskRunObject struct {
+	tr        *v1beta1.TaskRun
+	clientSet versioned.Interface
+	ctx       context.Context
+}
+
+func NewTaskRunObject(tr *v1beta1.TaskRun, clientSet versioned.Interface, ctx context.Context) *TaskRunObject {
+	return &TaskRunObject{
+		tr:        tr,
+		clientSet: clientSet,
+		ctx:       ctx,
+	}
+}
+
+func (tro *TaskRunObject) GetName() string {
+	return tro.tr.Name
+}
+
+func (tro *TaskRunObject) GetNamespace() string {
+	return tro.tr.Namespace
+}
+
+func (tro *TaskRunObject) GetKind() string {
+	return taskRunKind
+}
+
+func (tro *TaskRunObject) GetAnnotation(annotation string) *Annotation {
+	val, ok := tro.tr.Annotations[annotation]
+	return &Annotation{
+		Err:   nil,
+		Value: val,
+		Ok:    ok,
+	}
+}
+
+func (tro *TaskRunObject) GetLatestAnnotation(annotation string) *Annotation {
+	tr, err := tro.clientSet.TektonV1beta1().TaskRuns(tro.tr.Namespace).Get(tro.ctx, tro.tr.Name, v1.GetOptions{})
```

Comment:
I'd lean towards relying on Listers where possible (unless we absolutely need the freshest value here)

Reply:
Yup we were keeping that in mind. This call happens because we wanted to keep the original behavior, as of now the some storage options will make calls to the master API. [Here's an example.](https://github.com/tektoncd/chains/blob/main/pkg/chains/storage/tekton/tekton.go#L85)